### PR TITLE
Refactor(products endpoint) to consider user and favorites

### PR DESCRIPTION
**Pull request by Yuliia-Balandiuk**
# Description:

Refactor products endpoint to consider user and favorites

# Jira:
https://hackyourfuture-dk.atlassian.net/browse/CLASS22-141

### CODE REVIEW

File: packages/client/src/components/ProductList/ProductList.component.js [hunk 1]
```diff
@@ -8,8 +8,7 @@ export default function ProductList({
   isLoading,
   products,
   filteredProducts,
-  toggleFavorite,
-  favoriteProducts,
+  // toggleFavorite,
```
Comment (miltonhultgren) [inline packages/client/src/components/ProductList/ProductList.component.js]: Please don't comment out code, delete it if it isn't needed and if you change your mind, Git allows you to revert changes.

File: packages/client/src/containers/CollectionsPage/CollectionsPage.Container.js [hunk 2]
```diff
@@ -17,10 +16,7 @@ export const CollectionsPage = () => {
   const [categories, setCategories] = useState([]);
   const [searchInput, setSearchInput] = useState('');
   const [filteredProducts, setFilteredProducts] = useState([]);
-  const [isFavorite, setIsFavorite] = useState(false);
-  const [favoriteProducts, setFavoriteProducts] = useState(
-    getLocalStorage(favoriteProductsStorageKey) || [],
-  );
+  let isFavorite;
```
Comment (miltonhultgren) [inline packages/client/src/containers/CollectionsPage/CollectionsPage.Container.js:19]: This is never assigned any value?

File: packages/server/api/controllers/products.controller.js [hunk 3]
```diff
@@ -2,8 +2,18 @@ const knex = require('../../config/db');
 const HttpError = require('../lib/utils/http-error');
 
 // Get All products
-const getProducts = async () => {
-  return knex.select().table('products');
+const getProducts = async (token) => {
+  const userUid = token.split(' ')[1];
+  const user = (await knex('users').where({ uid: userUid }))[0];
```
Comment (miltonhultgren) [inline packages/server/api/controllers/products.controller.js:7]: If there is no such user found that should probably be an error thrown.

File: packages/server/api/controllers/users.controllers.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const knex = require('../../config/db');
+
+// Get All Users
+const getUsers = async () => {
```
Comment (miltonhultgren) [inline packages/server/api/controllers/users.controllers.js:4]: Where is this used?

**Review by miltonhultgren — CHANGES_REQUESTED** [review]

**Review by 27ManishaPatel — DISMISSED** [review]

**Pull request merged**